BrowserKit/Sources/WebEngine/WKWebview/Scripts/AdsTelemetryContentScript.swift

```diff
@@ -0,0 +1,114 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Foundation
+import WebKit
+
+public struct EngineSearchProviderModel {
+    typealias Predicate = (String) -> Bool
+    let name: String
+    let regexp: String
+    let queryParam: String
+    let codeParam: String
+    let codePrefixes: [String]
+    let followOnParams: [String]
+    let extraAdServersRegexps: [String]
+
+    public init(name: String,
+                regexp: String,
+                queryParam: String,
+                codeParam: String,
+                codePrefixes: [String],
+                followOnParams: [String],
+                extraAdServersRegexps: [String]) {
+        self.name = name
+        self.regexp = regexp
+        self.queryParam = queryParam
+        self.codeParam = codeParam
+        self.codePrefixes = codePrefixes
+        self.followOnParams = followOnParams
+        self.extraAdServersRegexps = extraAdServersRegexps
+    }
+}
+
+extension EngineSearchProviderModel {
```

Comment:
nit; does this need to be in an extension?

Reply:
No I will move it out 👍 , I was just following the previous pattern in the original code (which had this in an extension)